tests/resources/functions/ruby-3.1/tests.rb

```diff
@@ -1,36 +1,87 @@
-#        'variables' - object with function variables
-#    'res' variable has:
-#        'send(text, status)' - function to return text response. Status code defaults to 200
-#        'json(obj, status)' - function to return JSON response. Status code defaults to 200
-#    
-#    If an error is thrown, a response with code 500 will be returned.
+def main(context)
+    action = context.req.headers['x-action'] || ''
 
-def main(req, res)
-    payload = JSON.parse(req.payload === '' ? '{}' : req.payload)
+    case action
+    when 'plaintextResponse'
+        return context.res.send('Hello World 👋')
+    when 'jsonResponse'
+        return context.res.json({ 'json': true, 'message': 'Developers are awesome.' })
+    when 'redirectResponse'
+        return context.res.redirect('https://github.com/')
+    when 'emptyResponse'
+        return context.res.empty()
+    when 'noResponse'
+        context.res.send('This should be ignored, as it is not returned.')
+        return nil
+    when 'doubleResponse'
+        context.res.send('This should be ignored.')
+        return context.res.send('This should be returned.')
+    when 'headersResponse'
+        return context.res.send('OK', 200, {
+            'first-header': 'first-value',
+            'second-header': context.req.headers['x-open-runtimes-custom-in-header'] || 'missing',
+            'x-open-runtimes-custom-out-header': 'third-value'
+        })
+    when 'statusResponse'
+        return context.res.send('FAIL', 404)
+    when 'requestMethod'
+        return context.res.send(context.req.method)
+    when 'requestUrl'
+        return context.res.json({
+            'url': context.req.url,
+            'port': context.req.port,
+            'path': context.req.path,
+            'query': context.req.query,
+            'queryString': context.req.query_string,
+            'scheme': context.req.scheme,
+            'host': context.req.host,
+        })
+    when 'requestHeaders'
+        return context.res.json(context.req.headers)
+    when 'requestBodyPlaintext'
+        return context.res.send(context.req.body)
+    when 'requestBodyJson'
+        key1 = nil
+        key2 = nil
 
-    todo = JSON.parse(HTTParty.get("https://jsonplaceholder.typicode.com/todos/" + (payload['id'] || '1')).body)
+        if context.req.body.instance_of?(String)
+            key1 = 'Missing key'
+            key2 = 'Missing key'
+        else
+            key1 = context.req.body['key1'] || 'Missing key'
+            key2 = context.req.body['key2'] || 'Missing key'
+        end
 
-    puts 'String1'
-    puts 42
-    puts 4.2
-    puts true
+        return context.res.json({
+            'key1': key1,
+            'key2': key2,
+            'raw': context.req.body_string
+        })
+    when 'envVars'
+        return context.res.json({
+            'var': ENV['CUSTOM_ENV_VAR'] || nil,
+            'emptyVar': ENV['NOT_DEFINED_VAR'] || nil
+        })
+    when 'logs'
+        print('Native log')
```

Comment:
what happens if we use `puts` ? We should have a test case for that too

Reply:
I tried it manually and it works fine.
Do we really need test for it? In JS we would need like 5 more tests then, for all log types.

It would make tests inconsistent across languages and would need runtime-specific tests. Which I would strongly want to avoid.

Reply:
`puts` is more idiomatic Ruby, if anything I would test for that instead of `print`.

I would check a dump of the entire log too instead of only checking if it contains a string, I suspect you'll have extra newlines when using `puts` as it adds a newline, and print does not

Reply:
Updates test to use `puts`